deployment/network-operator/templates/_helpers.tpl

```diff
@@ -65,3 +65,9 @@ Selector labels
 app.kubernetes.io/name: {{ include "network-operator.name" . }}
 app.kubernetes.io/instance: {{ .Release.Name }}
 {{- end }}
+
+{{- define "network-operator.imagePullSecret" }}
```

Comment:
im thinking we should have a defined template for every component that supports image pull secret.
then in the relevant template (e.g operator, nicclusterpolicy) we just use the respective defined template

e.g

`network-operator.operator.imagePullSecrets`
`network-operator.sriovDevicePlugin.imagePullSecrets`
`network-operator.secondaryNetwork.multus.imagePullSecrets`

each will look roughly the same:

```
{{- define "network-operator.sriovDevicePlugin.imagePullSecrets" }}
{{- if .Values.sriovDevicePlugin.imagePullSecrets }}
.Values.sriovDevicePlugin.imagePullSecrets
{{- else }}
.Values.imagePullSecrets
{{- end}}
{{- end }}
```

then you just use that in the templates
e.g in nicclusterpolicy template, sriovDevicePlugin section
```
  sriovDevicePlugin:
    image: {{ .Values.sriovDevicePlugin.image }}
    repository: {{ .Values.sriovDevicePlugin.repository }}
    version: {{ .Values.sriovDevicePlugin.version }}
    imagePullSecrets: {{ include "network-operator.sriovDevicePlugin.imagePullSecrets" . | nindent 4 }}
```

if both the global and the per resource imagePullSecrets is not specified it would default to empty list (as expected)

note: didnt test the above so some tweaks may be needed :)

WDYT ?

Reply:
it moves all logic from CR template to some functions. It simplifies CR template which is more actively changing, so I like this proposal